### PR TITLE
caasp: add extra configs for apparmor profiles

### PR DESCRIPTION
**Pull request by Itxaka**
Some of our containers may need extra configs for apparmor.
For example, neutron-dhcp launches dnsmasq, which needs access
to the neutron config dir and the /var/lib/neutron/dhcp for leases.

We need a way of extending apparmor profiles with custom directives
for the proper execution of our openstack containers.

This patch allows us to add extra directives for apparmor profiles

### CODE REVIEW

File: playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+- name: Apparmor config
+  block:
+    # need to use shell because mount is not able to remount without passing a lot of params that we may not know
+    - name: Set the root disk into rw
+      shell: "mount -o remount, rw /"
```
Comment (rhafer) [inline playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: Why is this needed at all? Isn't /etc an overlay mounted rw by default?

File: playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Copy openstack apparmor profile into the workers
+  copy:
+    src: "{{ item }}"
+    dest: "/etc/apparmor.d/local/{{ item }}"
+  with_items: "{{ apparmor_profiles }}"
```
Comment (evrardjp) [inline playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: Can you use `loop` instead of `with_items` ?

File: playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  register: apparmor_copy
```
Comment (evrardjp) [inline playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: Can you use

`notify:`

and handlers instead?

File: playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml (hunk 2 above)
Comment (Itxaka) [reply playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: done

File: playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml (hunk 3 above)
Comment (Itxaka) [reply playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: nope, dynamic handlers do not work on ansible, nor does handlers with parameters, which kind of sucks

Comment (evrardjp) [reply playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: What do you mean by dynamic handlers do not work on ansible?

Comment (Itxaka) [reply playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: ok I did some hack around to manage those dynamic handlers, but IMO this is ugly AF.

Comment (Itxaka) [reply playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: > What do you mean by dynamic handlers do not work on ansible?

something like:

```
- name: restart service {{ item }}
  service:
    name: {{ item }}
    state: restarted

```

so then you can just notify with `restart service WHATEVER` and get it done dynamically.

My hack kind of works but its really ugly, I think handlers should work more like this.

Comment (evrardjp) [reply playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: Can be done with a chain of handlers and a set fact, but ok.

File: playbooks/roles/setup-caasp-workers/handlers/main.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+# ansible-lint: we need to reload the profiel and there is no ansible modules for that yet
+- name: Reload apparmor profile
+  shell: "apparmor_parser -r /etc/apparmor.d/{{ item.item }}"
+  when: item.changed
+  loop: "{{ apparmor_copy.results | flatten(levels=1) }}"
+  loop_control:
+    label: "{{ item.item }}"
```
Comment (evrardjp) [inline playbooks/roles/setup-caasp-workers/handlers/main.yml]: nit: Not sure why you would want to set the label to this, it doesn't improve readability.

File: playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml (hunk 3 above)
Comment (evrardjp) [reply playbooks/roles/setup-caasp-workers/tasks/apparmor-profiles.yml]: That was a nit: anyway.

File: playbooks/roles/setup-caasp-workers/handlers/main.yml (hunk 4 above)
Comment (evrardjp) [reply playbooks/roles/setup-caasp-workers/handlers/main.yml]: I misread I thought this was loop_var.
